Exported Android apk insta-crashing with APK expansion and PCK encryption

### Godot version

v4.0.3.stable.official [5222a99f5]

### System information

Windows 11, Android Samsung A14, Android SDK API 33

### Issue description

When using custom build export templates with Android, PCK encryption and APK expansion, the Android App is insta-crashing with:

```
 FATAL EXCEPTION: main
Process: org.philhudson.test_and_enc, PID: 23907
 java.lang.RuntimeException: Unable to start activity ComponentInfo{org.philhudson.test_and_enc/com.godot.game.GodotApp}: java.lang.IllegalArgumentException: org.philhudson.test_and_enc: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4169)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4325)
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2574)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:226)
                                                                                                    	at android.os.Looper.loop(Looper.java:313)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8757)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
                                                                                                    Caused by: java.lang.IllegalArgumentException: org.philhudson.test_and_enc: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
                                                                                                    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
                                                                                                    	at android.app.PendingIntent.checkFlags(PendingIntent.java:408)
                                                                                                    	at android.app.PendingIntent.getActivityAsUser(PendingIntent.java:491)
                                                                                                    	at android.app.PendingIntent.getActivity(PendingIntent.java:477)
                                                                                                    	at android.app.PendingIntent.getActivity(PendingIntent.java:441)
                                                                                                    	at org.godotengine.godot.Godot.onCreate(Godot.java:688)
                                                                                                    	at androidx.fragment.app.Fragment.performCreate(Fragment.java:2949)
                                                                                                    	at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:475)
                                                                                                    	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:278)
                                                                                                    	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
                                                                                                    	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
                                                                                                    	at androidx.fragment.app.FragmentManager.execSingleAction(FragmentManager.java:1971)
                                                                                                    	at androidx.fragment.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:311)
                                                                                                    	at org.godotengine.godot.FullScreenGodotApp.onCreate(FullScreenGodotApp.java:69)
                                                                                                    	at com.godot.game.GodotApp.onCreate(GodotApp.java:45)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:8591)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:8570)
                                                                                                    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1384)
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4150)
                                                                                                    	... 12 more
                                                                                                    
```

There have been suggestions to turn on Gradle build and add the following dependency, as per: 

https://stackoverflow.com/questions/67045607/how-to-resolve-missing-pendingintent-mutability-flag-lint-warning-in-android-a

and

https://github.com/godotengine/godot/issues/38758

adding: ` implementation "androidx.work:work-runtime-ktx:2.7.1"`

does not resolve the issue.

Also note that for the reproduction project, I am not using gradle build to show the most basic form of the issue.

Please let me know if you have additional comments or requests for additional reproduction steps.

### Steps to reproduce

1. Open up example project
2. Attempt to run on Android device or simulator with one click deploy or exported APK
3. Observe logs in adb, see app crash instantly on device

### Minimal reproduction project

Adding shared url as the project includes export templates which are big, total download size around 150mb which exceeds the 25mb limit here:

https://drive.google.com/file/d/10_ZnggmkBl39KIhkSyiEQaNiRrwN3Lkc/view?usp=sharing

--- a/platform/android/java/lib/src/org/godotengine/godot/Godot.java
+++ b/platform/android/java/lib/src/org/godotengine/godot/Godot.java
@@ -711,8 +711,14 @@ public class Godot extends Fragment implements SensorEventListener, IDownloaderC
 				Intent notifierIntent = new Intent(activity, activity.getClass());
 				notifierIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
 
-				PendingIntent pendingIntent = PendingIntent.getActivity(activity, 0,
-						notifierIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+				PendingIntent pendingIntent;
+				if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+					pendingIntent = PendingIntent.getActivity(activity, 0,
+							notifierIntent, PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
+				} else {
+					pendingIntent = PendingIntent.getActivity(activity, 0,
+							notifierIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+				}
 
 				int startResult;
 				try {

